src/rule_definition_tools.jl

```diff
@@ -236,11 +236,7 @@ end
 
 function standard_propagation_expr(Δs, ∂s)
     # This is basically Δs ⋅ ∂s
-
-    # Notice: the thunking of `∂s[i] (potentially) saves us some computation
-    # if `Δs[i]` is a `AbstractDifferential` otherwise it is computed as soon
-    # as the pullback is evaluated
-    ∂_mul_Δs = [:(@thunk($(∂s[i])) * $(Δs[i])) for i in 1:length(∂s)]
+    ∂_mul_Δs = ntuple(i->:($(∂s[i]) * $(Δs[i])), length(∂s))
```

Comment:
Does the thunk not save us compute in the case of multiplying by `One`?

Reply:
Oh you answered this already

> In general thunking around scalars is pointless since the costs of creating a closure is going to be greater than basically any scalar operation.